.github/workflows/lint.yaml

```diff
@@ -1,75 +1,106 @@
-      - name: golangci-lint (api)
-        uses: golangci/golangci-lint-action@v3
-        with:
-          version: ${{ env.GOLANGCI_LINT_VERSION }}
-          working-directory: api
-          skip-cache: true
-      - name: golangci-lint (teleport)
-        uses: golangci/golangci-lint-action@v3
-        with:
-          version: ${{ env.GOLANGCI_LINT_VERSION }}
-          args: --build-tags libfido2,piv
-          skip-cache: true
-      - name: golangci-lint (kube-agent-updater)
-        uses: golangci/golangci-lint-action@v3
+      
+  # Run various golangci-lint checks.
+  lint-go:
+    name: Lint Go code
+    container:
+      image: ghcr.io/gravitational/teleport-buildbox:teleport15
+    strategy:
+      # This is useful in case there are multiple different linting errors in a PR.
+      fail-fast: ${{ github.event_name == 'merge_group' }}
+      matrix:
+        linter-config:
+          - name: api
+            working-directory: api
+          - name: teleport
+            args: "--build-tags libfido2,piv"
+            runner: ubuntu-22.04-32core
+          - name: kube-agent-updater
+            working-directory: integrations/kube-agent-updater
+          - name: assetes/backport
+            working-directory: assets/backport
+          - name: build.assets/tooling
+            working-directory: build.assets/tooling
+    runs-on: ${{ matrix.linter-config.runner || 'ubuntu-22.04' }}
+    steps:
+      - name: Collect Workflow Telemetry
+        uses: runforesight/workflow-telemetry-action@v1
         with:
-          version: ${{ env.GOLANGCI_LINT_VERSION }}
-          working-directory: integrations/kube-agent-updater
-          skip-cache: true
-      - name: golangci-lint (assets/backport)
+          comment_on_pr: false
+          proc_trace_sys_enable: true
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Mark the source control directory as safe
+        run: git config --global --add safe.directory $( realpath . )
+      - name: golangci-lint (${{ matrix.linter-config.name }})
         uses: golangci/golangci-lint-action@v3
         with:
-          version: ${{ env.GOLANGCI_LINT_VERSION }}
-          working-directory: assets/backport
+          args: ${{ matrix.linter-config.args }}
           skip-cache: true
-      - name: golangci-lint (build.assets/tooling)
-        uses: golangci/golangci-lint-action@v3
-        with:
           version: ${{ env.GOLANGCI_LINT_VERSION }}
-          working-directory: build.assets/tooling
-          skip-cache: true
+          working-directory: ${{ matrix.linter-config.working-directory }}
 
+  lint-misc:
+    name: Lint miscelaneous files (Rust, Shell, Helm charts, license headers, etc.)
+    runs-on: ubuntu-22.04-16core
+    container:
+      image: ghcr.io/gravitational/teleport-buildbox:teleport15
+    steps:
+      - name: Collect Workflow Telemetry
+        uses: runforesight/workflow-telemetry-action@v1
```

Comment:
Can we pin all instances of `runforesight/workflow-telemetry-action`?  It is a pretty small dependency -- I don't particularly trust the maintainers long term.  Dependabot will keep it up to date automatically.

Reply:
Yea sure. Unfortunately it hasn't been updated in more than 6 months. Fingers crossed it hasn't been abandoned entirely... it's a really neat tool